rpcserver.go

```diff
@@ -732,6 +732,9 @@ func createVoutList(mtx *wire.MsgTx, chainParams *chaincfg.Params, filterAddrMap
 		var vout btcjson.Vout
 		vout.N = uint32(i)
 		vout.Value = btcutil.Amount(v.Value).ToBTC()
+		if len(encodedAddrs) == 1 && reqSigs == 1 {
```

Comment:
Are we sure the `reqSigs` is always set to 1 on old versions? Can't we just omit that condition? And I guess a comment above about why we do this (backward compatibility) would be useful too.

Reply:
Added the condition because if more than 1 signatures are required then we shouldn't consider the address to be the only address. This could result in terrible bugs, such as the consumer believing they have received the Bitcoin when it's actually in a multisig.

I'll double-check and can add a comment and make it also accept 0 as long as an address is available. The `reqSigs` should be correct but I agree it makes sense for this to be robust.

Reply:
Added a comment and changed the `reqSigs` check to also accept 0. `ExtractPkScriptAddrs` simply returns the `reqSigs` as 1 whenever an address is returned, except for multisig which has more logic for `reqSigs` and where checking the value of `reqSigs` is more interesting.

Reply:
Also checked that `ExtractPkScriptAddrs` returns a single address and `reqSigs == 1` for P2PKH, P2WPKH, P2SH, P2WSH, and P2TR.

The change would only be meaningful to non standard multi-sig.